app/instrumentation-tests/src/org/commcare/androidTests/FormEntryTest.kt

```diff
@@ -0,0 +1,300 @@
+        onView(withText("Unsent Forms: 1"))
+                .check(doesNotExist())
+        onView(withText("Start"))
+                .perform(click())
+        Espresso.pressBack()
+        onView(withText(startsWith("You last synced with the server:")))
+                .check(matches(isDisplayed()))
+
+        // Confirm form is present in saved forms
+        onView(withText("Saved"))
+                .perform(click())
+        onView(withText("Languages"))
+                .check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun testSaveCase() {
+        InstrumentationUtility.login("form_tests", "123")
+        // Create incomplete update case form.
+        InstrumentationUtility.openForm(1, 1)
+        openCase("Snow")
+        onView(withId(R.id.nav_btn_next))
+                .perform(click())
+        onView(withClassName(endsWith("EditText")))
+                .perform(typeText("hello"))
+        closeSoftKeyboard()
+        Espresso.pressBack()
+        onView(withText(R.string.keep_changes))
+                .perform(click())
+
+        // testing notification for having incomplete form for case already made
+        InstrumentationUtility.openForm(1, 1)
+        openCase("Snow")
+        confirmNotification_whenCaseHasIncompeleteForm()
+        onView(withText("NO"))
+                .perform(click())
+        onView(withText("A"))
+                .perform(click())
+        onView(withId(R.id.nav_btn_next))
+                .perform(click())
+        onView(withClassName(endsWith("EditText")))
+                .check(matches(withText(""))) // we don't see the text hello here.
+        closeSoftKeyboard()
+        Espresso.pressBack()
+        onView(withText(R.string.keep_changes))
+                .perform(click())
+
+        // Deleting one incomplete form for case
+        onView(withText(startsWith("Incomplete")))
+                .perform(click())
+        onView(CustomMatchers.find(
+                allOf(withText("Update a Case")),
+                1
+        )).perform(longClick())
+        onView(withText("Open"))
+                .check(matches(isDisplayed()))
+        onView(withText("Delete Record"))
+                .check(matches(isDisplayed()))
+        onView(withText("Scan Record Integrity"))
+                .check(matches(isDisplayed()))
+        onView(withText("Delete Record"))
+                .perform(click())
+
+        // Confirm we still have one and only one incomplete case.
+        onView(withText("Update a Case"))
+                .check(matches(isDisplayed()))
+        Espresso.pressBack()
+
+        // Continue incomplete case form.
+        InstrumentationUtility.openForm(1, 1)
+        openCase("Snow")
+        confirmNotification_whenCaseHasIncompeleteForm()
+        onView(withText("YES"))
+                .perform(click())
+        onView(withId(R.id.jumpEndButton))
+                .perform(click())
+
+        // Save form
+        onView(withText("hello"))
```

Comment:
not required

Reply:
Just this particualr check is not required? Or the part of saving the form as incomplete and checking it later is not required?

Reply:
the text entry as "hello" is not required since we are not opening and checking this value again in the test.

Reply:
Ohh this is being checked here to be present. It was entered earlier. So should I leave it or I understood it wrong?

Reply:
sorry To rephrase, even if you remove L268, this test would still pass right ?

Reply:
Yeah that's correct.

Reply:
yeah so I just meant this statement doesn't seem to serve anything meaningful in this test and therefore we should remove it.

Reply:
Though it isn't part of the original calabash test, but I think it's meaningful. 
When we go to update the case again, and press `YES` to the popup asking `Do you want to continue filling out the saved form`, so we should check whether it's really the form we saved first or is it a completely new one. And that check is done by this line.

Reply:
Ahh that makes sense. Sounds good to retain it.